pkg/crc/machine/bundle/repository.go

```diff
@@ -198,8 +229,14 @@ func Use(bundleName string) (*CrcBundleInfo, error) {
 	return defaultRepo.Use(bundleName)
 }
 
-func Extract(ctx context.Context, path string) (*CrcBundleInfo, error) {
-	if err := defaultRepo.Extract(ctx, path); err != nil {
+func Extract(ctx context.Context, reader io.Reader, path string) (*CrcBundleInfo, error) {
```

Comment:
We could use just one return statement and one error check in this case:

```go
var err error
if reader == nil {
	err = defaultRepo.Extract(ctx, path)
} else {
	err = defaultRepo.ExtractWithReader(ctx, reader, path)
}

if err != nil {
	return nil, err
}
return defaultRepo.Get(filepath.Base(path))
```

Maybe there could be some more blank lines too.